exercises/practice/dnd-character/tests/Tests.elm

```diff
@@ -0,0 +1,223 @@
+                <|
+                    \character ->
+                        Expect.all [ Expect.atLeast 3, Expect.atMost 18 ] character.intelligence
+            , skip <|
+                fuzzWith { runs = 100, distribution = Test.noDistribution }
+                    (Fuzz.fromGenerator DndCharacter.character)
+                    "generated character wisdom should be within range"
+                <|
+                    \character ->
+                        Expect.all [ Expect.atLeast 3, Expect.atMost 18 ] character.wisdom
+            , skip <|
+                fuzzWith { runs = 100, distribution = Test.noDistribution }
+                    (Fuzz.fromGenerator DndCharacter.character)
+                    "generated character charisma should be within range"
+                <|
+                    \character ->
+                        Expect.all [ Expect.atLeast 3, Expect.atMost 18 ] character.charisma
+            , skip <|
+                fuzzWith { runs = 100, distribution = Test.noDistribution }
+                    (Fuzz.fromGenerator DndCharacter.character)
+                    "generated character hitpoints should be 10 plus the constitution modifier"
+                <|
+                    \character ->
+                        Expect.equal character.hitpoints (10 + DndCharacter.modifier character.constitution)
+            , skip <|
+                fuzzWith { runs = 100, distribution = Test.noDistribution }
+                    (Fuzz.fromGenerator DndCharacter.character)
+                    "generated character abilities are not all equal"
+                <|
+                    \character ->
+                        let
+                            uniqueAbilities { strength, dexterity, constitution, intelligence, wisdom, charisma } =
+                                [ strength, dexterity, constitution, intelligence, wisdom, charisma ]
+                                    |> Set.fromList
+                                    |> Set.size
+                        in
+                        Expect.atLeast 2 (uniqueAbilities character)
+            , skip <|
+                fuzzWith
+                    { runs = 100
+                    , distribution =
+                        Test.expectDistribution
+                            [ ( Test.Distribution.atLeast 40, "has low charisma", \char -> char.charisma <= 12 )
+                            , ( Test.Distribution.atLeast 40, "has high charisma", \char -> char.charisma > 12 )
+                            , ( Test.Distribution.atLeast 40, "has low strength", \char -> char.strength <= 12 )
+                            , ( Test.Distribution.atLeast 40, "has high strength", \char -> char.strength > 12 )
+                            ]
+                    }
+                    (Fuzz.fromGenerator DndCharacter.character)
+                    "generated characters are not all equal"
+                <|
+                    \_ -> Expect.pass
```

Comment:
Finally, this one check that we don't always generate the same character.

I looked at the cumulative probabilities of each score, and there is a 51.2% chance of getting a 12 or less, so I required the distribution to have at least 40% of 1-12 and at least 40% of 13-18. The binomial distribution gave me a `10^{-12}` probability to have less than 40% in 1000 runs.

An error for a test of this type looks like
```
    Distribution report:
    ====================
      has low charisma:                    100%  (100x)  ██████████████████████████████
      has low strength:                    100%  (100x)  ██████████████████████████████
      has high charisma:                     0%    (0x)  ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
      has high strength:                     0%    (0x)  ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
    
    Combinations (included in the above base counts):
      has low charisma, has low strength:  100%  (100x)  ██████████████████████████████
    
    Distribution of label "has high charisma" was insufficient:
      expected:  40.000%
      got:       0.000%.
    
    (Generated 100 values.)
```